src/main/java/com/simprints/libsimprints/Registration.java

```diff
@@ -82,102 +54,47 @@ public int describeContents() {
     @Override
     public void writeToParcel(Parcel dest, int flags) {
         dest.writeString(guid);
-        if (rightIndex == null) {
-            dest.writeInt(0);
-            dest.writeByteArray(new byte[0]);
-        }
-        else {
-            dest.writeInt(rightIndex.length);
-            dest.writeByteArray(rightIndex);
-        }
-        if (rightThumb == null) {
-            dest.writeInt(0);
-            dest.writeByteArray(new byte[0]);
-        }
-        else {
-            dest.writeInt(rightThumb.length);
-            dest.writeByteArray(rightThumb);
-        }
-        if (leftIndex == null) {
-            dest.writeInt(0);
-            dest.writeByteArray(new byte[0]);
-        }
-        else {
-            dest.writeInt(leftIndex.length);
-            dest.writeByteArray(leftIndex);
-        }
-        if (leftThumb == null) {
-            dest.writeInt(0);
-            dest.writeByteArray(new byte[0]);
-        }
-        else {
-            dest.writeInt(this.leftThumb.length);
-            dest.writeByteArray(leftThumb);
+        dest.writeInt(templates.size());
+        for (Map.Entry<FingerIdentifier, byte[]> e : templates.entrySet()) {
+            dest.writeInt(e.getKey().ordinal());
+            dest.writeInt(e.getValue().length);
+            dest.writeByteArray(e.getValue());
         }
     }
 
     public String getGuid() {
         return guid;
     }
 
-    public void setGuid(String guid) {
-        this.guid = guid;
-    }
-
-    public byte[] getRightIndex() {
-        return rightIndex;
-    }
-
-    public void setRightIndex(byte[] rightIndex) {
-        this.rightIndex = rightIndex;
-    }
-
-    public byte[] getRightThumb() {
-        return rightThumb;
+    public void setTemplate(FingerIdentifier fingerId, byte[] fingerTemplate) {
```

Comment:
the `byte[] fingerTemplate` argument should have a `@NonNull` annotation or the [parcel code should check](https://github.com/SimPrints/LibSimprints/pull/3/files#diff-aecc8d0c67d5f8facf0cb77888ecb513R60) when the template is `null`.